### PR TITLE
Adding example

### PR DESCRIPTION
**Pull request by sirandreww**

### CODE REVIEW

**Comment by sirandreww** [comment]
this is meant to solve:
 Model saving in struct requires different type. #727 
 Get size of tensor #729

File: examples/13-housing-nn-in-struct.rs [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+// ************************************************************************************************
+// Model definition
+// ************************************************************************************************
+
+type Mlp = (
+    (Linear<2, 32>, ReLU),
+    (Linear<32, 32>, ReLU),
+    (Linear<32, 1>, Sigmoid),
+);
```
Comment (coreylowman) [inline examples/13-housing-nn-in-struct.rs:33]: Thoughts on adding a type alias for the built version?

```suggestion
type MlpStructure = (
    (Linear<2, 32>, ReLU),
    (Linear<32, 32>, ReLU),
    (Linear<32, 1>, Sigmoid),
);
type Mlp = <MlpStructure as BuildOnDevice<AutoDevice, f32>>::Built;
```

**Review by coreylowman — CHANGES_REQUESTED** [review]
Looks awesome! Left a couple of suggestions, but other than that this looks good to me.

File: examples/13-housing-nn-in-struct.rs [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+fn function_we_would_like_the_nn_to_mimic(
+    input: Tensor<(Const<2>,), f32, AutoDevice>,
+) -> Tensor<(Const<1>,), f32, AutoDevice> {
+    let distance_from_center: Tensor<(), f32, AutoDevice> = input.powi(2).sum().sqrt();
+    if distance_from_center.as_vec()[0] > 1.0 {
+        AutoDevice::default().tensor([1.0])
+    } else {
+        AutoDevice::default().tensor([0.0])
+    }
```
[the remark this answers is not in the document]
Comment (sirandreww) [reply examples/13-housing-nn-in-struct.rs:130]: Brilliant!

File: examples/13-housing-nn-in-struct.rs (hunk 1 above)
Comment (sirandreww) [reply examples/13-housing-nn-in-struct.rs:33]: Can you explain the reasoning behind this? what is the logical difference between MLP and MLPStructure ?

**Comment by coreylowman** [comment]
Thanks for the contribution!! 🔥

**Pull request merged**